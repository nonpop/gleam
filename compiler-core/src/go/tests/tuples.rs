use crate::assert_go;

#[test]
fn tuple() {
    assert_go!(
        r#"
fn go() {
  #("1", "2", "3")
}
"#,
    );
}

#[test]
fn tuple1() {
    assert_go!(
        r#"
fn go() {
  #(
    "1111111111111111111111111111111",
    #("1111111111111111111111111111111", "2", "3"),
    "3",
  )
}
"#,
    );
}

#[test]
fn tuple_access() {
    assert_go!(
        r#"
fn go() {
  #(1, 2).0
}
"#,
    )
}

#[test]
fn tuple_with_block_element() {
    assert_go!(
        r#"
fn go() {
  #(
    "1",
    {
      "2"
      "3"
    },
  )
}
"#,
    );
}

#[test]
fn tuple_with_block_element1() {
    assert_go!(
        r#"
fn go() {
  #(
    "1111111111111111111111111111111",
    #("1111111111111111111111111111111", "2", "3"),
    "3",
  )
}
"#,
    );
}

#[test]
fn constant_tuples() {
    assert_go!(
        r#"
const a = "Hello"
const b = 1
const c = 2.0
const e = #("bob", "dug")
        "#,
    );
}

#[test]
fn constant_tuples1() {
    assert_go!(
        r#"
const e = #(
  "loooooooooooooong", "loooooooooooong", "loooooooooooooong",
  "loooooooooooooong", "loooooooooooong", "loooooooooooooong",
)
"#
    );
}

#[test]
fn case() {
    assert_go!(
        r#"
fn go(a) {
  case a {
    #(2, a) -> a
    #(1, 1) -> 1
    #(a, b) -> a + b
  }
}
"#,
    );
}

#[test]
fn nested_pattern() {
    assert_go!(
        r#"
fn go(x) {
  case x {
    #(2, #(a, b)) -> a + b
    _ -> 1
  }
}
"#,
    );
}
