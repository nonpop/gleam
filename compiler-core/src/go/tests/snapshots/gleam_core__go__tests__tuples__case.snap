---
source: compiler-core/src/go/tests/tuples.rs
expression: "\nfn go(a) {\n  case a {\n    #(2, a) -> a\n    #(1, 1) -> 1\n    #(a, b) -> a + b\n  }\n}\n"
snapshot_kind: text
---
----- SOURCE CODE

fn go(a) {
  case a {
    #(2, a) -> a
    #(1, 1) -> 1
    #(a, b) -> a + b
  }
}


----- COMPILED GO
package mod_P

import (
  gleam_P "example.com/todo/gleam"
)

const _ = gleam_P.Use_Import
const Use_Import byte = 0

func goʹ(a gleam_P.Tuple2_t[gleam_P.Int_t, gleam_P.Int_t]) gleam_P.Int_t {
  if gleam_P.Bool_t(a.P_0.Equal(2)) {
    var aʹ1 gleam_P.Int_t = a.P_1
    _ = aʹ1
    return aʹ1
  } else if gleam_P.Bool_t(a.P_0.Equal(1)) && gleam_P.Bool_t(a.P_1.Equal(1)) {
    return 1
  } else {
    var aʹ1 gleam_P.Int_t = a.P_0
    _ = aʹ1
    var b gleam_P.Int_t = a.P_1
    _ = b
    return aʹ1 + b
  }
}
